### DISCUSSION TITLE
How to make `npm link -ws` like `npm install` without install packages

### DISCUSSION THREAD

**Discussion by Aqours** [npm/feedback#556] [CLI] [answered]
## Env

node: 16
npm: 7

## Monorepo Use Case

1. Install node_modules(public packages) inside my docker image.
2. Run docker container and mount volume which only include project source code.
3. Want to link monorepo packages(private packages) in monorepo root without `npm install`

### 1. npm link

```bash
cd monorepo
npm link -ws
````
 Not work. Error: Modules not found

### 2. npm link with prefix argument

```bash
cd monorepo
npm link -ws --prefix /monorepo
# OR
npm link -ws --prefix /monorepo --global
```
Not work.

### 3. npm link with prefix

```bash
cd monorepo
npm config -g set prefix /monorepo
npm link -ws
```

It works if I don't run `npm link -ws` before `npm config -g set prefix /monorepo`

## Question

How can I use `npm link` without `npm install` and `npm config set` to link monorepo packages?

**Comment by wraithgar** [comment, marked as answer]
I don't get that error when I run `npm link` inside a project with workspaces, but no `node_modules`.  Can you provide a more reliable way to reproduce this?

**Reply by Aqours** [reply]
emm... You're right. Actual behavior is there is no `node_modules`.

---

> Not work. Error: Modules not found

It's a `require` error.
